### PR TITLE
LUN-264: Locking lane for exercises screen

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/routes/exercises/components/LockingLane.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  justify-content: center;
+  margin: 10px;
+  max-height: ${hp('10%')}px;
+`
+
+const Line = styled.View<{ color: string }>`
+  width: 2px;
+  height: ${hp('5%')}px;
+  background-color: ${props => props.color};
+  margin: 10px;
```
Comment (f1sh1918) [inline src/routes/exercises/components/LockingLane.tsx:19]: pls use spacing from theme

File: src/routes/exercises/components/LockingLane.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  let Icon
+  if (current.level < index) {
+    Icon = LockIcon
+  } else if (current.level === index) {
+    Icon = CircleIconBlue
+  } else {
+    Icon = CheckCircleIconBlue
+  }
+
+  const colorPre = current.level < index ? 'black' : COLORS.lockingLane
```
Comment (f1sh1918) [inline src/routes/exercises/components/LockingLane.tsx:37]: i think all colors we use should be defined in `colors.ts`

File: src/routes/exercises/ExercisesScreen.tsx [hunk 3]
```diff
@@ -108,13 +123,16 @@ const ExercisesScreen = ({ route, navigation }: ExercisesScreenProps): JSX.Eleme
...
-          <FlatList
-            data={EXERCISES}
-            ListHeaderComponent={Header}
-            renderItem={Item}
-            keyExtractor={({ key }) => key.toString()}
-            showsVerticalScrollIndicator={false}
-          />
+          <>
+            <Title title={disciplineTitle} description={wordsDescription(documents.length)} />
+
```
Comment (f1sh1918) [inline src/routes/exercises/ExercisesScreen.tsx:128]: rm empty line

**Review by f1sh1918 — CHANGES_REQUESTED** [review]
Tested on ios. For me every exercise page shoes the same progress, even i didn't start the exercise

**Review by steffenkleinle — COMMENTED** [review]
Is something else required for this to work? It shows the checkmark for all exercises except from an empty circle for the write exercise for me. When I complete the write exercise, it stays the same. This is the case for ALL disciplines. Looking good except from that though, I like the design.

**Review by steffenkleinle — APPROVED** [review]

**Review by f1sh1918 — APPROVED** [review]
sorry i didn't realize that the logic gonna be implemented in a separate issue. so everything is fine :) nice work!

**Comment by SteffiStuffel** [comment]
> Is something else required for this to work? It shows the checkmark for all exercises except from an empty circle for the write exercise for me. When I complete the write exercise, it stays the same. This is the case for ALL disciplines. Looking good except from that though, I like the design.

This ticket is just for design. You can change hardCoded status in ExercisesScreen.tsx to see how it looks for no or one exercises finished. Logic will be done in next ticket.

**Pull request merged**